static/app/views/settings/account/accountSecurity/accountSecurityEnroll.tsx

```diff
@@ -330,6 +335,36 @@ class AccountSecurityEnroll extends DeprecatedAsyncView<Props, State> {
 
     this.props.router.push('/settings/account/security/');
     openRecoveryOptions({authenticatorName: this.authenticatorName});
+
+    // The remainder of this function is included primarily to smooth out the relocation flow. The
+    // newly claimed user will have landed on `https://sentry.sentry.io/settings/account/security`
+    // to perform the 2FA registration. But now that they have in fact registered, we want to
+    // redirect them to the subdomain of the organization they are already a member of (ex:
+    // `https://my-2fa-org.sentry.io`), but did not have the ability to access due to their previous
+    // lack of 2FA enrollment.
+    let orgs = OrganizationsStore.getAll();
+    if (orgs.length === 0) {
+      // Try to load orgs post 2FA again.
+      orgs = await fetchOrganizations(this.api, {member: '1'});
+      OrganizationsStore.load(orgs);
+
+      // Still no orgs? Nowhere to redirect the user to, so just stay in place.
+      if (orgs.length === 0) {
+        return;
+      }
+    }
+
+    // If we are already in an org sub-domain, we don't need to do any redirection. If we are not
+    // (this is usually only the case for a newly claimed relocated user), we redirect to the org
+    // slug's subdomain now.
+    const subDomains = getSubDomains(new URL(window.location.href));
+    const firstSubDomain = subDomains.length > 0 ? subDomains[0] : '';
+    const isAlreadyInOrgSubDomain = orgs.some(org => {
+      return org.slug === firstSubDomain;
+    });
+    if (!isAlreadyInOrgSubDomain) {
+      window.location.assign(generateOrgSlugUrl(orgs[0].slug));
+    }
```

Comment:
Organizations from the API should include a `links` property that has both `organizationUrl` and `regionUrl` for the organization. You could use those values to compare with the current host and skip splitting up subdomains.

Reply:
Switched to using `organizationUrl` as you suggest in the latest push.